src/tfr_sensor/src/light_detection_action_client.cpp

```diff
@@ -0,0 +1,32 @@
+#include <ros/ros.h>
+#include <ros/console.h>
+#include <actionlib/client/simple_action_client.h>
+#include <tfr_msgs/LightDetectAction.h>
+
+/**
+ *  Test code to never be used again, demonstrates light detection functionality
+ * */
+int main(int argc, char** argv)
+{
+    ros::init(argc, argv, "light_detection_action_client");
+    ros::NodeHandle n;
+    actionlib::SimpleActionClient<tfr_msgs::LightDetectAction> client(
+            "light_detection", true);
+
+    ROS_INFO("waiting for detection action server to start");
+    client.waitForServer();
+    ROS_INFO("client connected");
+    tfr_msgs::LightDetectGoal goal{};
+    
+    client.sendGoal(goal);
+    ros::Duration(0.5).sleep();
+    client.cancelAllGoals();
+    ros::Duration(0.5).sleep();
+
+    client.sendGoal(goal);
+    while (!client.waitForResult(ros::Duration(0.25)));
```

Comment:
Do we have anything to handle if we don't see the light for a very long time? While obviously this would be very bad for this to happen in a match (it means either our light hardware broke or we're searching in a veeery wrong place), it might be good to have a timeout (even if when it times out we just run it again and output a debug message) just in case so that we don't get stuck waiting forever? Just a thing I noticed and figured I might bring up.

Reply:
I'm thinking just adding a boolean to the action result saying whether we saw it or not?

Reply:
If the light circuit on the wall doesn't work we'll hit the wall and get disqualified.

I don't think we could conceivably end up in a position where we're backing in the wrong direction, because we don't start that routine unless we see the ArUco marker.

Reply:
I think this would be such a fatal error that we couldn't/don't want our system to attempt to recover. The user would need to recognize it and do manual control. 

When we integrate this with communication, we can make sure to put a message or status code letting the user know what is going on, but autonomous recovery is probably not desirable.